web/client/reducers/draw.js

```diff
@@ -49,7 +50,21 @@ function draw(state = initialState, action) {
             currentStyle: action.currentStyle
         });
     case GEOMETRY_CHANGED:
-        return assign({}, state, {tempFeatures: action.features});
+        let newData = action.features;
+        if (newData[0].geometry.type === 'Point') {
+            const normalizedPoint = [normalizeLng(newData[0].geometry.coordinates[0]), newData[0].geometry.coordinates[1]];
+            newData[0].geometry.coordinates = normalizedPoint;
+        } else if (newData[0].geometry.type === 'LineString') {
+            const normalizedLineString =  newData[0].geometry.coordinates.map((item) => [normalizeLng(item[0]), item[1]]);
+            newData[0].geometry.coordinates = normalizedLineString;
+        } else if (newData[0].geometry.type === 'MultiPoint') {
+            const normalizedMultiPoint =  newData[0].geometry.coordinates.map((item) => [normalizeLng(item[0]), item[1]]);
+            newData[0].geometry.coordinates = normalizedMultiPoint;
+        } else {
+            const normalizedData = newData[0].geometry.coordinates.map(i => i.map(x => x.map((item) => [normalizeLng(item[0]), item[1]])));
+            newData[0].geometry.coordinates = normalizedData;
+        }
```

Comment:
Polygon and Multilinestring has similar structure so you can check them together

but MultiPolygon should have one more level